### PR TITLE
PUB-1070 - Reference Table Conversion Routine

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1070

### Change description ###

When P&I data management API receives a artefact, it contains the provenance source and provenance source id. Using court reference table, we need to find P&I internal court Id and save the artefact against internal court Id.

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]
PR also needs switching to merge into demo rather than master

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java [hunk 1]
```diff
@@ -258,4 +270,13 @@ public void checkNewlyActiveArtefacts() {
         List<Artefact> newArtefactsToday = artefactRepository.findArtefactsByDisplayFrom(LocalDate.now());
         newArtefactsToday.forEach(artefact -> log.info(sendArtefactForSubscription(artefact)));
     }
+
+    public void findByCourtIdByProvenanceAndUpdate(Artefact artefact) {
+        Optional<Court> court = courtRepository.findByCourtIdByProvenance(artefact.getProvenance(), artefact.getCourtId());
+        if (!court.isEmpty()) {
+            artefact.setCourtId(court.get().getCourtId().toString());
+        } else {
+            throw new CourtNotFoundException(String.format("NoMatch: %s", artefact.getCourtId()));
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java]: As discussed, it's done.

**Review by furnivall — DISMISSED** [review]
👍 Looks good to me, assuming the new suppression isn't a problem :)

File: src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java [hunk 2]
```diff
@@ -40,12 +43,15 @@
 import static uk.gov.hmcts.reform.pip.data.management.helpers.TestConstants.MESSAGES_MATCH;
 
 @ExtendWith(MockitoExtension.class)
-@SuppressWarnings("PMD.TooManyMethods")
+@SuppressWarnings("PMD")
```
Comment (ChrisS1512) [inline src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java:46]: Is line 46 still needed?

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationTest.java [hunk 3]
```diff
@@ -1789,8 +1789,6 @@ void retrieveMetadataWhenVerificationEndpointNotSet() throws Exception {
             exceptionResponse.getMessage().contains(VERIFICATION_HEADER),
             "Verification error not shown in error message"
         );
-        assertTrue(exceptionResponse.getMessage().contains(VERIFICATION_HEADER),
-                   "Verification error not shown in error message");
     }
```
Comment (ChrisS1512) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationTest.java:1792]: We should add some integration tests for the NoMatch logic, if the court doesn't exist

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java]: fixed.

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationTest.java (hunk 3 above)
Comment (junaidiqbalmoj) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationTest.java:1792]: as discussed, add integration tests.

**Review by alberto-hortiguela — DISMISSED** [review]

**Review by furnivall — DISMISSED** [review]

**Review by alberto-hortiguela — APPROVED** [review]

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java [hunk 4]
```diff
@@ -258,4 +269,14 @@ public void checkNewlyActiveArtefacts() {
         List<Artefact> newArtefactsToday = artefactRepository.findArtefactsByDisplayFrom(LocalDate.now());
         newArtefactsToday.forEach(artefact -> log.info(sendArtefactForSubscription(artefact)));
     }
+
+    private void findByCourtIdByProvenanceAndUpdate(Artefact artefact) {
+        Optional<List<Court>> courts = courtRepository.findByCourtIdByProvenance(artefact.getProvenance(),
+                                                                          artefact.getCourtId());
+        if (courts.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:279]: As discussed, change has been made.

**Review by ChrisS1512 — APPROVED** [review]

**Review by alberto-hortiguela — DISMISSED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by furnivall — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**